src/go/k8s/cmd/main.go

```diff
@@ -388,7 +389,12 @@ func main() {
 
 			cacheRecorder := helmSourceController.MustMakeCacheMetrics()
 			indexTTL := 15 * time.Minute
-			helmIndexCache := helmSourceController.NewCache(0, indexTTL)
+
+			var helmIndexCache helmSourceController.Cache
+			if helmCacheMaxSize > 0 {
```

Comment:
I don't think we need this conditional?

Reply:
yes we do, If we are set to 0 we will get a false positive for an error.

Reply:
Can you add a note saying as much in the code? 😄 

Would you mind noting what the behavior is if `helmIndexCache` is `nil`? Uninitialized interfaces _generally_ means nil pointer exceptions later on.